### ISSUE TITLE
cigarette_smokers example fails suddenly at the end

### ISSUE THREAD

**Issue by digitalextremist** [celluloid/celluloid-fsm#1]
Running the example results in this, before or after EVERYONE SMOKES!
It's not always before, it's not always after. Technically the example works, except for this:

```
E, [2015-04-11T05:46:46.676000 #1864] ERROR -- : Actor crashed!
LocalJumpError: yield called out of block
	/home/de/.rvm/gems/jruby-1.7.19/bundler/gems/celluloid-b766f4b162b5/lib/celluloid/actor.rb:357:in `task'
	/home/de/.rvm/gems/jruby-1.7.19/bundler/gems/celluloid-b766f4b162b5/lib/celluloid/task.rb:57:in `initialize'
	/home/de/.rvm/gems/jruby-1.7.19/bundler/gems/celluloid-b766f4b162b5/lib/celluloid/task.rb:47:in `initialize'
	/home/de/.rvm/gems/jruby-1.7.19/bundler/gems/celluloid-b766f4b162b5/lib/celluloid/task/fibered.rb:15:in `create'
```

**Comment by tarcieri** [comment]
If the example actually worked, it'd keep running in a loop... why is it broken?

**Comment by digitalextremist** [comment]
jruby: `LocalJumpError: no block given (yield)`
rbx: `LocalJumpError: no block given`
mri ( 2.2.0 ): `LocalJumpError: yield called out of block`

I'm looking through this. It's an easier way to trigger an error like it I'd been getting.

**Comment by e2** [comment]
@digitalextremist, @tarcieri bisect points to: https://github.com/celluloid/celluloid/commit/06710a55a2790bafb88c3363636a4677ef60dd4d

Needs a unit test asap.

**Comment by e2** [comment]
Here's the issue: https://github.com/celluloid/celluloid/issues/554

**Comment by digitalextremist** [comment]
@e2 yep. That has sent a deep crack through the foundation.

**Comment by digitalextremist** [comment]
This is verified solved as of celluloid/celluloid#575

**Comment by e2** [comment]
Hurray! The smokers can once again enjoy their carcinogens to the fullest!
